File: json-rpc-server/api/src/main/java/dev/cloudgt/matlab/MatlabEngineApi.java
package dev.cloudgt.matlab;

import java.io.Writer;

public interface MatlabEngineApi {
    void eval(String command, Writer stdout, Writer stderr) throws
            InterruptedException,
            IllegalStateException,
            MatlabExecutionException,
            MatlabSyntaxException;

    Object feval(int nlhs, String func, Writer stdout, Writer stderr, Object... args) throws
            InterruptedException,
            IllegalStateException,
            MatlabExecutionException,
            UnsupportedTypeException,
            MatlabSyntaxException;

    Object getVariable(String name) throws
            InterruptedException,
            IllegalStateException;

    void putVariable(String name, Object value) throws
            InterruptedException,
            IllegalStateException;

    void disconnect();

    class EngineException extends Exception {
        public EngineException(String message) {
            super(message);
        }
    }

    class MatlabExecutionException extends Exception {}

    class MatlabSyntaxException extends Exception {}

    class UnsupportedTypeException extends Exception {}
}

File: json-rpc-server/src/main/kotlin/dev/cloudgt/matlab/MatlabEngineApi.kt
package dev.cloudgt.matlab

import java.io.Writer

interface MatlabEngineApi {
    fun eval(string: String, stdout: Writer, stderr: Writer)
    fun <T> feval(nlhs: Int, func: String, stdout: Writer, stderr: Writer, vararg args: Any): T?
    fun <T> getVariable(name: String): T?
    fun putVariable(name: String, value: Any)
    fun disconnect()
}